tests/testthat/test-rng.R

```diff
@@ -988,3 +988,179 @@ test_that("We can compile the standalone program", {
   cmp <- colSums(dust_rng$new(42, 5)$uniform(10, 0, 1))
   expect_equal(ans, cmp)
 })
+
+
+## Testing for the hypergeometric is a little different to above as it
+## turns out that despite the very different underlying
+## implementations, the algorithm in the paper
+## (https://www.tandfonline.com/doi/abs/10.1080/00949658508810839) has
+## been followed pretty exactly. So if we were willing to assume that
+## the R version is correct we can just check that the stream of
+## numbers is exactly the same.
+##
+## We do this twice - once between R's rhyper and a direct
+## reimplementation of the Rust code in R (within helper-rng) using
+## R's RNG stream, then again between our R reimplementation and
+## dust's idiomatic C++ version.
```

Comment:
why? why not just test the C++ version directly against rhyper?

Reply:
because they use different random number sources - we can't get rhyper to use dust's stream, and can't get dust to use R's stream. I wrote the reference version to accept any source of uniform random numbers to act as a bridge there